mllib/src/main/scala/org/apache/spark/mllib/linalg/Matrices.scala

```diff
@@ -179,14 +179,12 @@ private[spark] class MatrixUDT extends UserDefinedType[Matrix] {
         val tpe = row.getByte(0)
         val numRows = row.getInt(1)
         val numCols = row.getInt(2)
-        val values = row.getAs[Seq[Double]](5, ArrayType(DoubleType, containsNull = false)).toArray
+        val values = row.getArray(5).toArray.map(_.asInstanceOf[Double])
```

Comment:
actually i think in general we should have specialized getters even for toArray, and just call then. under the hood we can store them as objects and optimize them later, but it'd make the caller easier.